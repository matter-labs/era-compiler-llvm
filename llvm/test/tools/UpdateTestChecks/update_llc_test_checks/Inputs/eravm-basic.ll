; RUN: llc < %s -mtriple=eravm-unknown-unknown | FileCheck %s

@val = addrspace(4) global i256 42

define void @test1(i256 %rs1) {
  %valptr = alloca i256
  %val = load i256, i256 addrspace(4)* @val
  %res = add i256 %rs1, %val
  store i256 %res, i256* %valptr
  ret void
}

define void @test2(i256 %rs1) {
  %valptr = alloca i256
  %destptr = alloca i256
  %val = load i256, i256* %valptr
  %res = add i256 %val, %rs1
  store i256 %res, i256* %destptr
  ret void
}
