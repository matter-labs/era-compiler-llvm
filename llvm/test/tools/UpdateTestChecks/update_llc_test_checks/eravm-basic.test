# REQUIRES: eravm-registered-target

# RUN: cp -f %S/Inputs/eravm-basic.ll %t.ll && %update_llc_test_checks %t.ll
# RUN: diff -u %S/Inputs/eravm-basic.ll.expected %t.ll
