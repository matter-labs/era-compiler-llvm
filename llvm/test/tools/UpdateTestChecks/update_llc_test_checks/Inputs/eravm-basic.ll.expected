; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
; RUN: llc < %s -mtriple=eravm-unknown-unknown | FileCheck %s

@val = addrspace(4) global i256 42

define void @test1(i256 %rs1) {
; CHECK-LABEL: test1:
; CHECK:       ; %bb.0:
; CHECK-NEXT:    nop stack+=[1]
; CHECK-NEXT:    add @val[0], r1, stack-[1]
; CHECK-NEXT:    ret
  %valptr = alloca i256
  %val = load i256, i256 addrspace(4)* @val
  %res = add i256 %rs1, %val
  store i256 %res, i256* %valptr
  ret void
}

define void @test2(i256 %rs1) {
; CHECK-LABEL: test2:
; CHECK:       ; %bb.0:
; CHECK-NEXT:    nop stack+=[2]
; CHECK-NEXT:    add stack-[2], r1, stack-[1]
; CHECK-NEXT:    ret
  %valptr = alloca i256
  %destptr = alloca i256
  %val = load i256, i256* %valptr
  %res = add i256 %val, %rs1
  store i256 %res, i256* %destptr
  ret void
}
